src/core/src/FabricMeshManager.cpp

```diff
@@ -0,0 +1,213 @@
+#include "cesium/omniverse/FabricMeshManager.h"
+
+#include "cesium/omniverse/FabricGeometry.h"
+#include "cesium/omniverse/FabricGeometryDefinition.h"
+#include "cesium/omniverse/FabricGeometryPool.h"
+#include "cesium/omniverse/FabricMaterial.h"
+#include "cesium/omniverse/FabricMaterialDefinition.h"
+#include "cesium/omniverse/FabricMaterialPool.h"
+#include "cesium/omniverse/GltfUtil.h"
+
+#include <spdlog/fmt/fmt.h>
+
+namespace cesium::omniverse {
+
+std::shared_ptr<FabricMesh> FabricMeshManager::acquireMesh(
+    int64_t tilesetId,
+    int64_t tileId,
+    const glm::dmat4& ecefToUsdTransform,
+    const glm::dmat4& gltfToEcefTransform,
+    const glm::dmat4& nodeTransform,
+    const CesiumGltf::Model& model,
+    const CesiumGltf::MeshPrimitive& primitive,
+    bool smoothNormals,
+    const CesiumGltf::ImageCesium* imagery,
+    const glm::dvec2& imageryTexcoordTranslation,
+    const glm::dvec2& imageryTexcoordScale,
+    uint64_t imageryTexcoordSetIndex) {
+
+    const auto hasImagery = imagery != nullptr;
+    const auto geometry = acquireGeometry(model, primitive, smoothNormals, imagery, imageryTexcoordSetIndex);
+
+    geometry->setTile(
+        tilesetId,
+        tileId,
+        ecefToUsdTransform,
+        gltfToEcefTransform,
+        nodeTransform,
+        model,
+        primitive,
+        smoothNormals,
+        hasImagery,
+        imageryTexcoordTranslation,
+        imageryTexcoordScale,
+        imageryTexcoordSetIndex);
+
+    const auto hasMaterial = geometry->getGeometryDefinition().hasMaterial();
+    auto material = std::shared_ptr<FabricMaterial>(nullptr);
+
+    if (hasMaterial) {
+        material = acquireMaterial(model, primitive, imagery);
+
+        material->setTile(tilesetId, tileId, model, primitive, imagery);
+
+        geometry->assignMaterial(material);
+    }
+
+    return std::make_shared<FabricMesh>(geometry, material);
+}
+
+void FabricMeshManager::releaseMesh(std::shared_ptr<FabricMesh> mesh) {
+    const auto geometry = mesh->getGeometry();
+    const auto material = mesh->getMaterial();
+
+    assert(geometry != nullptr);
+
+    releaseGeometry(geometry);
+
+    if (material != nullptr) {
+        releaseMaterial(material);
+    }
+}
+
+uint64_t FabricMeshManager::getNumberOfGeometriesInUse() const {
+    uint64_t count = 0;
+    for (const auto geometryPool : _geometryPools) {
```

Comment:
All of these loops perform copies. These should be references.

```suggestion
    for (const auto &geometryPool : _geometryPools) {
```